File: src/application/enums/EquipmentType.java
package application.enums;

import application.models.Equipment;

public enum EquipmentType {

	SWORD, AXE, SABER,KNIFE, ROD, STAFF, HAND, 
	MAIL, ARMOR,  ROBE, CLOAK, 
	SHIELD, CROWN, AMULET, MASK, HELMET, MISC, EMPTY;


	public static EquipmentType getType(Equipment type)
	{
		if(type.getName().isEmpty()){return EMPTY;}
		switch(type.getName().substring(0, 2))
		{
		//Weapons
		case "SW":  return SWORD;
		case "AX":  return AXE;
		case "SB":  return SABER;
		case "KN":  return KNIFE;
		case "RD":  return ROD;
		case "ST":  return STAFF;
		case "HA":  return HAND;

		//Armors
		case "ML":  return MAIL;
		case "AR":  return ARMOR;
		case "CK":  return CLOAK;
		case "RB":  return ROBE;

		//Accessories
		case "SH":  return SHIELD;
		case "CR":  return CROWN;
		case "AM":  return AMULET;
		case "MK":  return MASK;
		case "HT":  return HELMET;

		default : return MISC;
		}
	}

	public static int compareTo(Equipment e1, Equipment e2)
	{
		return getType(e1).compareTo(getType(e2));
	}

}

File: src/application/enums/EquipmentType.kt
package application.enums

import application.models.Equipment

enum class EquipmentType
{

	SWORD, AXE, SABER, KNIFE, ROD,
	STAFF, HAND, MAIL, ARMOR, ROBE,
	CLOAK, SHIELD, CROWN, AMULET, MASK,
	HELMET, MISC, EMPTY;

	companion object
	{
		fun getType(type: Equipment): EquipmentType
		{
			if (type.name.isEmpty())
			{
				return EMPTY
			}
			when (type.name.substring(0, 2))
			{
			//Weapons
				"SW" -> return SWORD
				"AX" -> return AXE
				"SB" -> return SABER
				"KN" -> return KNIFE
				"RD" -> return ROD
				"ST" -> return STAFF
				"HA" -> return HAND

			//Armors
				"ML" -> return MAIL
				"AR" -> return ARMOR
				"CK" -> return CLOAK
				"RB" -> return ROBE

			//Accessories
				"SH" -> return SHIELD
				"CR" -> return CROWN
				"AM" -> return AMULET
				"MK" -> return MASK
				"HT" -> return HELMET
				else -> return MISC
			}
		}

		fun compareTo(e1: Equipment, e2: Equipment): Int
		{
			return getType(e1).compareTo(getType(e2))
		}
	}
}
